### PR TITLE
EVG-2911 Always check latest commit in smoke test

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: operations/service_deploy_smoke_utils.go [hunk 1]
```diff
@@ -68,7 +68,32 @@ func (tests smokeEndpointTestDefinitions) checkEndpoints() error {
...
+	defer resp.Body.Close()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get latest commit from GitHub")
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.Wrap(err, "error reading response body from GitHub")
+	}
+
+	latest := struct {
```
Comment (richardsamuels) [inline operations/service_deploy_smoke_utils.go]: I would recommend using github.Reference from go-github for this.

File: operations/service_deploy_smoke_utils.go [hunk 2]
```diff
@@ -68,7 +68,32 @@ func (tests smokeEndpointTestDefinitions) checkEndpoints() error {
 	return errors.Wrapf(catcher.Resolve(), "failed to get %d endpoints", catcher.Len())
 }
 
-func checkTaskByCommit(username, key, commit string) error {
+func getLatestGithubCommit() (string, error) {
+	client := util.GetHttpClient()
+	defer util.PutHttpClient(client)
+
+	resp, err := client.Get("https://api.github.com/repos/evergreen-ci/evergreen/git/refs/heads/master")
+	defer resp.Body.Close()
```
Comment (richardsamuels) [inline operations/service_deploy_smoke_utils.go]: Do this after err to avoid panic.

Comment (richardsamuels) [reply operations/service_deploy_smoke_utils.go]: Actually, that's still wrong. Do this right here:
```
if resp != nil {
        defer resp.Body.Close()
}
```

Comment (bsamek) [reply operations/service_deploy_smoke_utils.go]: Fixed. I think defer after err check if correct. The example from the docs is
```
resp, err := http.Get("http://example.com/")
if err != nil {
	// handle error
}
defer resp.Body.Close()
body, err := ioutil.ReadAll(resp.Body)
```
https://golang.org/pkg/net/http/

File: operations/service_deploy_smoke_utils.go (hunk 1 above)
Comment (bsamek) [reply operations/service_deploy_smoke_utils.go]: Done.

**Comment by bsamek** [comment]
@richardsamuels:

> What happens if another commit is made after the call to getLatestGithubCommit?

That's fine. If there is no build on a given hash, the loop continues, running `getLatestGithubCommit` again and checking that new hash.

**Pull request closed without merging**